### ISSUE TITLE
pyscan doesn't report quart 0.19.6 vulnerability CVE-2024-49767

### ISSUE THREAD

**Issue by oneumyvakin** [ohaswin/pyscan#20]
**Describe the bug**
There is a quart 0.19.6 vulnerability https://advisories.gitlab.com/pkg/pypi/quart/CVE-2024-49767/

**To Reproduce**
1. Create a file dir/pyproject.toml

```
[project]
dependencies = [
  "quart==0.19.6",
  "werkzeug==3.0.4",
  "starlette==0.38.2",
]
```
2. Run  pyscan -d dir/

**Actual behavior**
There is no vulnerability for quart:
```
pyscan -d pyscan/
pyscan v0.1.6 | by Aswin S (github.com/aswinnnn)
Using pyproject.toml as source...
Found 3 dependencies
|-| werkzeug [3.0.4] -> Found vulnerabilities!
|-| werkzeug [3.0.4] -> Found vulnerabilities!
|-| starlette [0.38.2] -> Found vulnerabilities!
|-| quart [0.19.6] -> No vulnerabilities found.
```

**Expected behavior**
There is the vulnerability for quart https://advisories.gitlab.com/pkg/pypi/quart/CVE-2024-49767/

**Additional information**

1. Create a file dir/pyproject.toml

```
[project]
dependencies = [
  "quart==0.19.6",
]
```
2. Run  pyscan -d dir/

```
export RUST_BACKTRACE=full
pyscan -d pyscan/
pyscan v0.1.6 | by Aswin S (github.com/aswinnnn)
Using pyproject.toml as source...
Found 1 dependencies
Found 1 vulnerabilities so far
thread 'main' panicked at src/scanner/models.rs:416:63:
called `Option::unwrap()` on a `None` value
stack backtrace:
   0:     0x61082775d4f5 - std::backtrace_rs::backtrace::libunwind::trace::h58eed11393533053
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/../../backtrace/src/backtrace/libunwind.rs:116:5
   1:     0x61082775d4f5 - std::backtrace_rs::backtrace::trace_unsynchronized::h6af9bae28ebb6388
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x61082775d4f5 - std::sys_common::backtrace::_print_fmt::hb6748916642a4fb2
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys_common/backtrace.rs:68:5
   3:     0x61082775d4f5 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h3692694645b1bb6a
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x610827787a8b - core::fmt::rt::Argument::fmt::h7aa93977ba74ae0f
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/fmt/rt.rs:165:63
   5:     0x610827787a8b - core::fmt::write::h5131d80b4c69b88d
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/fmt/mod.rs:1168:21
   6:     0x61082775a3ff - std::io::Write::write_fmt::h1fb327a7d8b0eb36
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/io/mod.rs:1835:15
   7:     0x61082775d2ce - std::sys_common::backtrace::_print::he6ebb7b9d89f4456
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x61082775d2ce - std::sys_common::backtrace::print::h998d75b840f75a73
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x61082775e669 - std::panicking::default_hook::{{closure}}::h18ec7fe6a38b9da0
  10:     0x61082775e40a - std::panicking::default_hook::hfb3f22c2e4075a6a
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:298:9
  11:     0x61082775eb03 - std::panicking::rust_panic_with_hook::h51af00bcb4660c4e
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:795:13
  12:     0x61082775e9ab - std::panicking::begin_panic_handler::{{closure}}::h39f76aa863fbe8ce
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:656:13
  13:     0x61082775d9b9 - std::sys_common::backtrace::__rust_end_short_backtrace::h4d10fc2251b89840
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys_common/backtrace.rs:171:18
  14:     0x61082775e717 - rust_begin_unwind
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:652:5
  15:     0x610826e18ce3 - core::panicking::panic_fmt::h319840fcbcd912ef
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/panicking.rs:72:14
  16:     0x610826e18d8c - core::panicking::panic::h19def44c80243eda
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/panicking.rs:146:5
  17:     0x610826e18c49 - core::option::unwrap_failed::h9b45086d3ec3e03c
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/option.rs:1985:5
  18:     0x610826e3936c - core::option::Option<T>::unwrap::h9236527cb6032a33
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/option.rs:935:21
  19:     0x610826e3936c - pyscan::scanner::models::Vulnerability::to_scanned_dependency::hc28ad9dd9c941cd7
                               at /home/aswin/projects/pyscan/src/scanner/models.rs:416:32
  20:     0x610826ea155a - pyscan::scanner::api::Osv::query_batched::{{closure}}::h9c26829b38b4e19c
                               at /home/aswin/projects/pyscan/src/scanner/api.rs:152:42
  21:     0x610826e3a8f5 - pyscan::scanner::start::{{closure}}::h740fd46fceaf6ad5
                               at /home/aswin/projects/pyscan/src/scanner/mod.rs:18:48
  22:     0x610826eaeea7 - pyscan::parser::find_pyproject_imports::{{closure}}::h13363899fd6ad7e5
                               at /home/aswin/projects/pyscan/src/parser/mod.rs:187:29
  23:     0x610826eacdc0 - pyscan::parser::find_import::{{closure}}::ha12ba533b26e537e
                               at /home/aswin/projects/pyscan/src/parser/mod.rs:87:40
  24:     0x610826eac67e - pyscan::parser::scan_dir::{{closure}}::hd91c98a0d31c3c29
                               at /home/aswin/projects/pyscan/src/parser/mod.rs:70:25
  25:     0x610826e96809 - pyscan::main::{{closure}}::hdd5d5312f4b55930
                               at /home/aswin/projects/pyscan/src/main.rs:184:41
  26:     0x610826e4f0cf - tokio::runtime::park::CachedParkThread::block_on::{{closure}}::hf5ca6a7d9b57f592
                               at /home/aswin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.1/src/runtime/park.rs:283:63
  27:     0x610826e4ea9c - tokio::runtime::coop::with_budget::h6f9c7f9b22967f56
                               at /home/aswin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.1/src/runtime/coop.rs:107:5
  28:     0x610826e4ea9c - tokio::runtime::coop::budget::h1abd1176e303196a
                               at /home/aswin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.1/src/runtime/coop.rs:73:5
  29:     0x610826e4ea9c - tokio::runtime::park::CachedParkThread::block_on::h7043b9527f8150fc
                               at /home/aswin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.1/src/runtime/park.rs:283:31
  30:     0x610826e37aa3 - tokio::runtime::context::BlockingRegionGuard::block_on::h400b1a01652a34e9
                               at /home/aswin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.1/src/runtime/context.rs:315:13
  31:     0x610826e7d91f - tokio::runtime::scheduler::multi_thread::MultiThread::block_on::h8d39522bd68460d2
                               at /home/aswin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.1/src/runtime/scheduler/multi_thread/mod.rs:66:9
  32:     0x610826e76cba - tokio::runtime::runtime::Runtime::block_on::hdb52039df7156b09
                               at /home/aswin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.1/src/runtime/runtime.rs:304:45
  33:     0x610826e61c49 - pyscan::main::hb41d7fb3c52e8e02
                               at /home/aswin/projects/pyscan/src/main.rs:183:5
  34:     0x610826e8b14b - core::ops::function::FnOnce::call_once::h515823d658ca4be0
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/ops/function.rs:250:5
  35:     0x610826e1ed8e - std::sys_common::backtrace::__rust_begin_short_backtrace::h174dc9b91ee0a57e
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys_common/backtrace.rs:155:18
  36:     0x610826e883f1 - std::rt::lang_start::{{closure}}::heafbb7ce3d1846b8
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/rt.rs:159:18
  37:     0x610827753d8d - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::hb84c63630a35bb05
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/ops/function.rs:284:13
  38:     0x610827753d8d - std::panicking::try::do_call::h8d62108d97b3e028
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:559:40
  39:     0x610827753d8d - std::panicking::try::h0c0b9a214b9691f1
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:523:19
  40:     0x610827753d8d - std::panic::catch_unwind::ha8912f28da143edb
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panic.rs:149:14
  41:     0x610827753d8d - std::rt::lang_start_internal::{{closure}}::h2baf77487fc7f90d
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/rt.rs:141:48
  42:     0x610827753d8d - std::panicking::try::do_call::h36fdc82521d3343f
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:559:40
  43:     0x610827753d8d - std::panicking::try::h58aa1415c41e30ec
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:523:19
  44:     0x610827753d8d - std::panic::catch_unwind::h3c4b5a8c3b1c4acf
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panic.rs:149:14
  45:     0x610827753d8d - std::rt::lang_start_internal::h63a185b0ddd212e9
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/rt.rs:141:20
  46:     0x610826e883ca - std::rt::lang_start::h4cbf630cfbd9c008
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/rt.rs:158:17
  47:     0x610826e61cee - main
  48:     0x706e8be29d90 - <unknown>
  49:     0x706e8be29e40 - __libc_start_main
  50:     0x610826e19811 - _start
  51:                0x0 - <unknown>
```

**Comment by ohaswin** [comment]
The API query works, but for some reason it responds with the package name "werkzeug" instead of quart, which led to the crash due to faulty logic. A fix has been implemented and will be released shortly.

**Comment by ohaswin** [comment]
Fixed in v0.1.7
